extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/WindowOperatorQueryFrameProcessor.java

```diff
@@ -99,6 +98,8 @@ public class WindowOperatorQueryFrameProcessor implements FrameProcessor<Object>
   private final ArrayList<ResultRow> rowsToProcess;
   private int lastPartitionIndex = -1;
 
+  final AtomicInteger rowId = new AtomicInteger(0);
```

Comment:
> and not set rowId to 0 anywhere

It would have to be done in the class as well.

> create a class for the frame writing process

It's getting difficult to generalize things into a common flow, since we have 2 different flows right now - (1) empty over() flow, and (2) non-empty over() flow.

I think the GlueingPartitioningOperator PR work would allow us to create a common class for the frame writing process. Can we take it up there? It would make it more messy and difficult to understand if we try to do it in this PR in my opinion.